File: Ent_UsAdmin/ActInformacionLegal.vb
Imports System.ComponentModel
Imports System.Data.SqlClient

Public Class ActInformacionLegal
    Public idCreditoLegal As String
    Dim domicilioActual As String
    Dim noExpedienteActual As String
    Dim JuzgadoActual As String
    Dim EtapaProcesalActual As String
    Dim ActuarioActual As String
    Dim consultaActualizacion As String
    Dim idSolicitud As String
    Dim idDatosSolicitud As String
    Dim NuevoDomicilio As String
    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub ComboTipo_onItemSelected(sender As Object, e As EventArgs) Handles ComboTipo.onItemSelected
        Select Case ComboTipo.selectedValue
            Case "Domicilio"
                PanelDomicilio.Location = New Point(41, 124)
                PanelDomicilio.Visible = True
                PanelValor.Visible = False
            Case Else
                PanelValor.Location = New Point(41, 124)
                PanelValor.Visible = True
                PanelDomicilio.Visible = False
        End Select
    End Sub

    Private Sub ActInformacionLegal_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        CheckForIllegalCrossThreadCalls = False
        Cargando.Show()
        Cargando.MonoFlat_Label1.Text = "Cargando Información"
        BackgroundConsultaInformacion.RunWorkerAsync()

    End Sub

    Private Sub BackgroundConsultaInformacion_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundConsultaInformacion.DoWork
        iniciarconexionempresa()

        Dim comandoInformacion As SqlCommand
        Dim consultaInformacion As String
        Dim readerInformacion As SqlDataReader
        consultaInformacion = "select NoExpediente,Juzgado,EtapaProcesal,Actuario,datosSolicitud.id as idDatosSolicitud from legales inner join Solicitud on Legales.idSolicitud = Solicitud.id inner join DatosSolicitud on Solicitud.id = DatosSolicitud.IdSolicitud where legales.id= '" & idCreditoLegal & "'"
        comandoInformacion = New SqlCommand
        comandoInformacion.Connection = conexionempresa
        comandoInformacion.CommandText = consultaInformacion
        readerInformacion = comandoInformacion.ExecuteReader
        If readerInformacion.HasRows Then
            While readerInformacion.Read
                noExpedienteActual = readerInformacion("Noexpediente")
                JuzgadoActual = readerInformacion("Juzgado")
                EtapaProcesalActual = readerInformacion("EtapaProcesal")
                ActuarioActual = readerInformacion("Actuario")
                idDatosSolicitud = readerInformacion("idDatosSolicitud")
            End While
        End If
        readerInformacion.Close()

        Dim comandoDomicilio As SqlCommand
        Dim consultaDomicilio As String
        consultaDomicilio = "select concat('Calle ',DatosSolicitud.Calle,' Número Exterior ',DatosSolicitud.Noext,' Número Interior ',DatosSolicitud.Noint,' Colonia ',DatosSolicitud.Colonia,' Código Postal ',DatosSolicitud.CodigoPostal) as Domicilio from Legales inner join Solicitud on legales.idSolicitud = Solicitud.id inner join DatosSolicitud on Solicitud.id = DatosSolicitud.IdSolicitud where legales.id = '" & idCreditoLegal & "'"
        comandoDomicilio = New SqlCommand
        comandoDomicilio.Connection = conexionempresa
        comandoDomicilio.CommandText = consultaDomicilio
        domicilioActual = comandoDomicilio.ExecuteScalar

    End Sub

    Private Sub BackgroundConsultaInformacion_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundConsultaInformacion.RunWorkerCompleted
        Cargando.Close()
    End Sub

    Private Sub BackgroundActualizacion_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundActualizacion.DoWork

        Dim tiempo As String = TimeOfDay.ToString("HH:mm:ss")
        Select Case ComboTipo.selectedValue
            Case "Domicilio"
                consultaActualizacion = "update datosSolicitud set calle = '" & txtCalle.Text & "',noext = '" & txtNoExt.Text & "',noint = '" & txtNoInt.Text & "',codigopostal = '" & txtCodigoPostal.Text & "',colonia = '" & txtColonia.Text & "' where id = '" & idDatosSolicitud & "'"
            Case "Número de Expediente"
                consultaActualizacion = "update legales set NoExpediente = '" & txtValor.Text & "' where id = '" & idCreditoLegal & "'"
            Case "Juzgado"
                consultaActualizacion = "update legales set Juzgado = '" & txtValor.Text & "' where id = '" & idCreditoLegal & "'"
            Case "Etapa Procesal"
                consultaActualizacion = "update legales set EtapaProcesal = '" & txtValor.Text & "' where id = '" & idCreditoLegal & "'"
            Case "Actuario"
                consultaActualizacion = "update legales set actuario = '" & txtValor.Text & "' where id = '" & idCreditoLegal & "'"
        End Select
        Dim comandoLegalAnterior As SqlCommand
        Dim consultaLegalAnterior As String

        Select Case ComboTipo.selectedValue
            Case "Domicilio"
                NuevoDomicilio = "Calle " & txtCalle.Text & " Número Exterior " & txtNoExt.Text & " Número Interior " & txtNoInt.Text & " Colonia " & txtColonia.Text & " Código Postal " & txtCodigoPostal.Text
                consultaLegalAnterior = "insert into ActualizacionesLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" & domicilioActual & "','" & NuevoDomicilio & "','" & txtMotivo.Text & "','" & idCreditoLegal & "')"
            Case "Número de Expediente"
                consultaLegalAnterior = "insert into ActualizacionesLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" & noExpedienteActual & "','" & txtValor.Text & "','" & txtMotivo.Text & "','" & idCreditoLegal & "')"
            Case "Juzgado"
                consultaLegalAnterior = "insert into ActualizacionesLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" & JuzgadoActual & "','" & txtValor.Text & "','" & txtMotivo.Text & "','" & idCreditoLegal & "')"
            Case "Etapa Procesal"
                consultaLegalAnterior = "insert into ActualizacionesLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" & EtapaProcesalActual & "','" & txtValor.Text & "','" & txtMotivo.Text & "','" & idCreditoLegal & "')"
            Case "Actuario"
                consultaLegalAnterior = "insert into ActualizacionesLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" & ActuarioActual & "','" & txtValor.Text & "','" & txtMotivo.Text & "','" & idCreditoLegal & "')"
        End Select
        'consultaLegalAnterior = "insert into ActualizacionLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" &  & "')"
        comandoLegalAnterior = New SqlCommand
        comandoLegalAnterior.Connection = conexionempresa
        comandoLegalAnterior.CommandText = consultaLegalAnterior
        comandoLegalAnterior.ExecuteNonQuery()


        Dim comandoAct As SqlCommand
        comandoAct = New SqlCommand
        comandoAct.Connection = conexionempresa
        comandoAct.CommandText = consultaActualizacion
        comandoAct.ExecuteNonQuery()




    End Sub

    Private Sub BunifuThinButton21_Click(sender As Object, e As EventArgs) Handles BunifuThinButton21.Click
        Cargando.Show()
        Cargando.MonoFlat_Label1.Text = "Actualizando Información"
        Cargando.TopMost = True
        BackgroundActualizacion.RunWorkerAsync()

    End Sub

    Private Sub BackgroundActualizacion_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundActualizacion.RunWorkerCompleted
        Cargando.Close()
        Me.Close()
    End Sub
End Class

File: Ent_UsAdmin/Legal/ActInformacionLegal.cs
using System;
using System.ComponentModel;
using System.Data.SqlClient;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class ActInformacionLegal
    {
        public string idCreditoLegal;
        private string domicilioActual;
        private string noExpedienteActual;
        private string JuzgadoActual;
        private string EtapaProcesalActual;
        private string ActuarioActual;
        private string consultaActualizacion;
        private string idSolicitud;
        private string idDatosSolicitud;
        private string NuevoDomicilio;

        public ActInformacionLegal()
        {
            InitializeComponent();
        }
        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void ComboTipo_onItemSelected(object sender, EventArgs e)
        {
            switch (ComboTipo.selectedValue ?? "")
            {
                case "Domicilio":
                    {
                        PanelDomicilio.Location = new Point(41, 124);
                        PanelDomicilio.Visible = true;
                        PanelValor.Visible = false;
                        break;
                    }

                default:
                    {
                        PanelValor.Location = new Point(41, 124);
                        PanelValor.Visible = true;
                        PanelDomicilio.Visible = false;
                        break;
                    }
            }
        }

        private void ActInformacionLegal_Load(object sender, EventArgs e)
        {
            CheckForIllegalCrossThreadCalls = false;
            My.MyProject.Forms.Cargando.Show();
            My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Cargando Información";
            BackgroundConsultaInformacion.RunWorkerAsync();

        }

        private void BackgroundConsultaInformacion_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();

            SqlCommand comandoInformacion;
            string consultaInformacion;
            SqlDataReader readerInformacion;
            consultaInformacion = "select NoExpediente,Juzgado,EtapaProcesal,Actuario,datosSolicitud.id as idDatosSolicitud from legales inner join Solicitud on Legales.idSolicitud = Solicitud.id inner join DatosSolicitud on Solicitud.id = DatosSolicitud.IdSolicitud where legales.id= '" + idCreditoLegal + "'";
            comandoInformacion = new SqlCommand();
            comandoInformacion.Connection = Module1.conexionempresa;
            comandoInformacion.CommandText = consultaInformacion;
            readerInformacion = comandoInformacion.ExecuteReader();
            if (readerInformacion.HasRows)
            {
                while (readerInformacion.Read())
                {
                    noExpedienteActual = Conversions.ToString(readerInformacion["Noexpediente"]);
                    JuzgadoActual = Conversions.ToString(readerInformacion["Juzgado"]);
                    EtapaProcesalActual = Conversions.ToString(readerInformacion["EtapaProcesal"]);
                    ActuarioActual = Conversions.ToString(readerInformacion["Actuario"]);
                    idDatosSolicitud = Conversions.ToString(readerInformacion["idDatosSolicitud"]);
                }
            }
            readerInformacion.Close();

            SqlCommand comandoDomicilio;
            string consultaDomicilio;
            consultaDomicilio = "select concat('Calle ',DatosSolicitud.Calle,' Número Exterior ',DatosSolicitud.Noext,' Número Interior ',DatosSolicitud.Noint,' Colonia ',DatosSolicitud.Colonia,' Código Postal ',DatosSolicitud.CodigoPostal) as Domicilio from Legales inner join Solicitud on legales.idSolicitud = Solicitud.id inner join DatosSolicitud on Solicitud.id = DatosSolicitud.IdSolicitud where legales.id = '" + idCreditoLegal + "'";
            comandoDomicilio = new SqlCommand();
            comandoDomicilio.Connection = Module1.conexionempresa;
            comandoDomicilio.CommandText = consultaDomicilio;
            domicilioActual = Conversions.ToString(comandoDomicilio.ExecuteScalar());

        }

        private void BackgroundConsultaInformacion_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            My.MyProject.Forms.Cargando.Close();
        }

        private void BackgroundActualizacion_DoWork(object sender, DoWorkEventArgs e)
        {

            string tiempo = DateAndTime.TimeOfDay.ToString("HH:mm:ss");
            switch (ComboTipo.selectedValue ?? "")
            {
                case "Domicilio":
                    {
                        consultaActualizacion = "update datosSolicitud set calle = '" + txtCalle.Text + "',noext = '" + txtNoExt.Text + "',noint = '" + txtNoInt.Text + "',codigopostal = '" + txtCodigoPostal.Text + "',colonia = '" + txtColonia.Text + "' where id = '" + idDatosSolicitud + "'";
                        break;
                    }
                case "Número de Expediente":
                    {
                        consultaActualizacion = "update legales set NoExpediente = '" + txtValor.Text + "' where id = '" + idCreditoLegal + "'";
                        break;
                    }
                case "Juzgado":
                    {
                        consultaActualizacion = "update legales set Juzgado = '" + txtValor.Text + "' where id = '" + idCreditoLegal + "'";
                        break;
                    }
                case "Etapa Procesal":
                    {
                        consultaActualizacion = "update legales set EtapaProcesal = '" + txtValor.Text + "' where id = '" + idCreditoLegal + "'";
                        break;
                    }
                case "Actuario":
                    {
                        consultaActualizacion = "update legales set actuario = '" + txtValor.Text + "' where id = '" + idCreditoLegal + "'";
                        break;
                    }
            }
            SqlCommand comandoLegalAnterior;
            string consultaLegalAnterior ="";

            switch (ComboTipo.selectedValue ?? "")
            {
                case "Domicilio":
                    {
                        NuevoDomicilio = "Calle " + txtCalle.Text + " Número Exterior " + txtNoExt.Text + " Número Interior " + txtNoInt.Text + " Colonia " + txtColonia.Text + " Código Postal " + txtCodigoPostal.Text;
                        consultaLegalAnterior = "insert into ActualizacionesLegal values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + tiempo + "','" + ComboTipo.selectedValue + "','" + domicilioActual + "','" + NuevoDomicilio + "','" + txtMotivo.Text + "','" + idCreditoLegal + "')";
                        break;
                    }
                case "Número de Expediente":
                    {
                        consultaLegalAnterior = "insert into ActualizacionesLegal values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + tiempo + "','" + ComboTipo.selectedValue + "','" + noExpedienteActual + "','" + txtValor.Text + "','" + txtMotivo.Text + "','" + idCreditoLegal + "')";
                        break;
                    }
                case "Juzgado":
                    {
                        consultaLegalAnterior = "insert into ActualizacionesLegal values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + tiempo + "','" + ComboTipo.selectedValue + "','" + JuzgadoActual + "','" + txtValor.Text + "','" + txtMotivo.Text + "','" + idCreditoLegal + "')";
                        break;
                    }
                case "Etapa Procesal":
                    {
                        consultaLegalAnterior = "insert into ActualizacionesLegal values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + tiempo + "','" + ComboTipo.selectedValue + "','" + EtapaProcesalActual + "','" + txtValor.Text + "','" + txtMotivo.Text + "','" + idCreditoLegal + "')";
                        break;
                    }
                case "Actuario":
                    {
                        consultaLegalAnterior = "insert into ActualizacionesLegal values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + tiempo + "','" + ComboTipo.selectedValue + "','" + ActuarioActual + "','" + txtValor.Text + "','" + txtMotivo.Text + "','" + idCreditoLegal + "')";
                        break;
                    }
            }
            // consultaLegalAnterior = "insert into ActualizacionLegal values('" & Now.ToString("yyyy-MM-dd") & "','" & tiempo & "','" & ComboTipo.selectedValue & "','" &  & "')"
            comandoLegalAnterior = new SqlCommand();
            comandoLegalAnterior.Connection = Module1.conexionempresa;
            comandoLegalAnterior.CommandText = consultaLegalAnterior;
            comandoLegalAnterior.ExecuteNonQuery();


            SqlCommand comandoAct;
            comandoAct = new SqlCommand();
            comandoAct.Connection = Module1.conexionempresa;
            comandoAct.CommandText = consultaActualizacion;
            comandoAct.ExecuteNonQuery();




        }

        private void BunifuThinButton21_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Cargando.Show();
            My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Actualizando Información";
            My.MyProject.Forms.Cargando.TopMost = true;
            BackgroundActualizacion.RunWorkerAsync();

        }

        private void BackgroundActualizacion_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            My.MyProject.Forms.Cargando.Close();
            Close();
        }
    }
}